runtime/compiler/z/codegen/UncommonBCDCHKAddressNode.cpp

```diff
@@ -62,17 +62,44 @@ UncommonBCDCHKAddressNode::perform()
             {
             TR::Node* pdOpNode = node->getFirstChild();
             TR::Node* oldAddressNode = node->getSecondChild();
-            TR_ASSERT(pdOpNode && oldAddressNode, "Unexpected null PD opNode or address node under BCDCHK");
-            TR_ASSERT(oldAddressNode->getNumChildren() == 2, "Expecting 2 children under address node of BCDCHK.");
+            TR_ASSERT_FATAL_WITH_NODE(node, pdOpNode && oldAddressNode, "Unexpected null PD opNode or address node under BCDCHK");
```

Comment:
I understand that this is the old code, but if the intention of this assert is that both `oldAddressNode` and `pdOpNode` should be non  null, then we should explicitly check that.